### PR TITLE
Strip leading spaces from compiler flags when building OpenSSL on Linux

### PR DESCRIPTION
**Pull request by sjdrc**
* I've checked this [Git style guide](https://0.readthedocs.io/en/latest/git.html). **[Yes]**
* I've checked this [CMake style guide](https://0.readthedocs.io/en/latest/cmake.html). **[Yes]**
* My change will work with CMake 3.2 (minimum requirement for Hunter). **[Yes]**
* I will try to keep this pull request as small as possible and will try not to mix unrelated features. **[Yes]**

---

### CODE REVIEW

File: cmake/projects/OpenSSL/schemes/url_sha1_openssl.cmake.in [hunk 1]
```diff
@@ -108,6 +108,8 @@ elseif(OPENWRT)
   set(configure_opts "linux-generic32" "no-async")
 endif()
 
+string(REGEX REPLACE "^ " "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
```
Comment (ruslo) [inline cmake/projects/OpenSSL/schemes/url_sha1_openssl.cmake.in]: Please update `PACKAGE_INTERNAL_DEPS_ID` for OpenSSL so we can test this change. See:
- https://docs.hunter.sh/en/latest/reference/user-modules/hunter_download.html

**Comment by sjdrc** [comment]
Done. Sorry for the delay.

**Comment by ruslo** [comment]
Testing:
- https://ci.appveyor.com/project/ingenue/hunter/builds/25086231
- https://travis-ci.org/ingenue/hunter/builds/542192457?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.188

**Pull request merged**